cf-agent/verify_environments.c

```diff
@@ -158,35 +158,37 @@ PromiseResult VerifyEnvironmentsPromise(EvalContext *ctx, const Promise *pp)
 
 /*****************************************************************************/
 
-static int EnvironmentsSanityChecks(Attributes a, const Promise *pp)
+static int EnvironmentsSanityChecks(const Attributes *a, const Promise *pp)
 {
-    if (a.env.spec)
+    assert(a != NULL);
+    const Environments env = a->env;
```

Comment:
Do we want to create this copy?

Reply:
Environments struct is relatively small (10 ints/pointers). I don't think this is a big deal, but using a pointer or copying just the parts needed are just as valid, imo.